autokeras/task.py

```diff
@@ -333,3 +333,92 @@ def __init__(self,
             max_trials=max_trials,
             directory=directory,
             seed=seed)
+
+
+class TimeSeriesForecaster(auto_model.AutoModel):
+    """AutoKeras time series data forecast class.
+
+    # Arguments
+        column_names: A list of strings specifying the names of the columns. The
+            length of the list should be equal to the number of columns of the data.
+            Defaults to None. If None, it will obtained from the header of the csv
+            file or the pandas.DataFrame.
+        column_types: Dict. The keys are the column names. The values should either
+            be 'numerical' or 'categorical', indicating the type of that column.
+            Defaults to None. If not None, the column_names need to be specified.
+            If None, it will be inferred from the data.
+        in_steps: Int. Number of lag observations as input (x). Values may be between
```

Comment:
There are many ways to convert a time series into a fixed number of steps: in particular using strides and using exponential moving averages. Same for the output. This type of decision should be left to the AutoML algorithm.

The question that is relevant to the user is "how far in the past should we look" and "how far in the future should we predict". But just looking "up to 100 timesteps in the past" doesn't necessarily mean that the model will have an input length of 100 (e.g. you could be using strides, etc).

So I'd suggest a `lookback` argument (integer, maximum age of input data in timesteps) and a `predict_until` argument. We could also offer more flexibility and have e.g. `predict_from=1` and `predict_until=10`.

Reply:
1. The user do not need to specify the column_names and column_types if the data are already vectorized. Actually, we only need the column_names if (1) the data are raw data and (2) column_names is not in the data and (3) the user is using the column_names to specify the types of the data in the column_types.

2. The meta_model would infer the column semantic types by looking into the values of the data.

Reply:
Agree. So with `predict_from` the user can have a time gap between the past and the predicted? If it doesn't affect the AutoML choices, i.e. we would need to predict all the values in the time gap before the specified time range then cut out the gap part before output, I think we should leave it to the user to do the cut.